main.nf

```diff
@@ -572,81 +539,42 @@ process generate_strain_tsv_vcf {
 
 }
 
-process somalier_extract {
-
-    input:
-        tuple path(vcf), file(vcf_index)
 
-    output:
-        path("*.somalier")
 
-    container 'brentp/somalier:dev'
+process generate_severity_tracks {
+    /*
+        The severity tracks are bedfiles with annotations for
+        LOW
+        MODERATE
+        HIGH
+        MODIFIER
+        variants as annotated with SNPEff
 
-    """
-        somalier extract --sites ${vcf} -f ${params.reference} ${vcf}
-    """
+        They are used on the CeNDR browser.
+    */
 
-}
+    publishDir "${params.output}/tracks", mode: 'copy'
 
-process somalier_relate {
-    
-    publishDir "${params.output}/somalier", mode: "copy"
+    tag { severity }
 
     input:
-        path("*.somalier")
-
+        tuple path("in.vcf.gz"), path("in.vcf.gz.csi"), val(severity)
     output:
-        path("somalier.pairs.tsv")
-        path("somalier.html")
-        path("somalier.pairs.tsv")
-        path("somalier.samples.tsv")
-
-    container 'brentp/somalier:dev'
+        set file("${date}.${severity}.bed.gz"), file("${date}.${severity}.bed.gz.tbi")
 
     """
-        somalier relate *.somalier
-    """
-
-}
-
-
-process calculate_gtcheck {
-
-    publishDir "${params.output}/concordance", mode: 'copy'
-
-    input:
-        tuple path(vcf), path(vcf_index)
-
-    output:
-        path("gtcheck.${date}.tsv")
-
-    """
-        {
-            echo -e "discordance\\tsites\\tavg_min_depth\\ti\\tj";
-            bcftools gtcheck -H -G 1 ${vcf} | egrep '^CN' | cut -f 2-6;
-        } > gtcheck.${date}.tsv
+        bcftools view --apply-filters PASS in.vcf.gz | \
+        grep ${severity} | \
+        awk '\$0 !~ "^#" { print \$1 "\\t" (\$2 - 1) "\\t" (\$2)  "\\t" \$1 ":" \$2 "\\t0\\t+"  "\\t" \$2 - 1 "\\t" \$2 "\\t0\\t1\\t1\\t0" }' | \\
+        bgzip  > ${date}.${severity}.bed.gz
+        tabix -p bed ${date}.${severity}.bed.gz
+        fsize=\$(zcat ${date}.${severity}.bed.gz | wc -c)
+        if [ \${fsize} -lt 2000 ]; then
+            exit 1
+        fi;
     """
 }
 
-process tajima_bed {
-
```

Comment:
Are you integrating this into the popgen pipeline?

Reply:
Yes I think so.